### PR TITLE
Don't block unix sockets

### PR DESCRIPTION
**Pull request by joetsoi**
This is a pretty dumb implementation and I'm happy to rework it to be a bit safer/defensive, just wanted to test if it worked first. It fixes issues with `asyncio` by check if it's an `AF_UNIX` domain socket being requested and allowing that, otherwise it blocks as normal.

should help with

https://github.com/miketheman/pytest-socket/issues/47
https://github.com/miketheman/pytest-socket/issues/6

I've tested it with both the starlette example and the `asynctest` example and both work with this patch

```
import asynctest

class TestClass(asynctest.TestCase):
    async def test_something(self):
        pass
```

### CODE REVIEW

**Comment by alex-verve** [comment]
Tested on 5 django projects and it works like a charm 👍 
I don't need to patch django via env var anymore, thanks for your work!

image

**Comment by miketheman** [comment]
Excellent, thanks for taking this on! I think this was recently enabled by converting this to a class, so yay!

I'm currently moving, so haven't had a change to check it out more thoroughly, but could you consider adding your test case to the tests so the functionality is exercised by CI?

Also, had you considered making this a configuration flag like --allow-unix-sockets so that the behavior can be controlled by the user? This way it's opt-in, and not a change of the current default behavior.

**Comment by joetsoi** [comment]
happy to do both! just wanted to throw this out there before investing too much time in it

**Comment by joetsoi** [comment]
Windows failure looks like an unrelated failure on setup. Not the absence of AF_UNIX that I was hoping to test

**Comment by joetsoi** [comment]
@miketheman, no pressure, just a ping hoping you find some time to take a look at this.

File: tests/test_socket.py [hunk 1]
```diff
@@ -228,3 +229,35 @@ class MySocket(socket.socket):
...
+
+        def test_unix_socket():
+            socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    """)
+    result = testdir.runpytest("--verbose", "--disable-socket")
+    if hasattr(socket, 'AF_UNIX'):
+        assert_socket_blocked(result)
+    else:
+        # no AF_UNIX on windows
+        result.assert_outcomes(passed=0, skipped=0, failed=1)
```
Comment (miketheman) [inline tests/test_socket.py:246]: **suggestion:** ‏I'm not a huge fan of conditional logic in tests, an improvement might be to use `@pytest.mark.skipif` to selectively skip these tests if the platform doesn't have unix sockets.
Right now the test is basically saying "I assume if you use AF_UNIX on Windows, the code fails" - and that's beyond the responsibility of this plugin, right?

Ref: https://docs.pytest.org/en/stable/reference.html#pytest-mark-skipif

If I'm reading this wrong, let me know!

**Review by miketheman — COMMENTED** [review]
Thanks for adding tests and a toggle! Change requested in the test logic - see inline!

**Comment by miketheman** [comment]
I also reran the test suite and the failing Windows test passed, so that's something we should keep an eye on - nothing worse than a flaky test!

**Review by miketheman — APPROVED** [review]

**Comment by miketheman** [comment]
Thanks for sticking with it!
I’ll try to get to a release next week.

**Pull request merged**